### PR TITLE
Extend and improve the 2.0 migration guide

### PR DESCRIPTION
**Pull request by blowekamp**
Added more sections, improved organization and simplified some
examples.

### CODE REVIEW

File: docs/source/migrationGuide2.0.rst [hunk 1]
```diff
@@ -1,34 +1,69 @@
...
 
 Migration Guide 2.0
--------------------
+===================
 
 With the release of SimpleITK 2.0 there are changes to the interface and to the behaviour of some methods and classes. The intent of this document is to provide guidance on how to modify code so that it is compatible with SimpleITK 2.0.
 
-DICOM reading
-+++++++++++++
+This document will continue to evolve after the SimpleITK 2.0 release, consider reading the `latest <https://simpleitk.readthedocs.io/en/master/migrationGuide2.0.html>`__ version. Contributions to update this document that share problems and solution when updating to SimpleITK 2.0 are welcomed.
```
Comment (zivy) [inline docs/source/migrationGuide2.0.rst:9]: solution -> solutions

File: docs/source/migrationGuide2.0.rst [hunk 2]
```diff
@@ -1,34 +1,69 @@
...
+| .. code-block:: python                            | .. code-block:: python                         |
+|                                                   |                                                |
+|  tx = sitk.Transform()                            |  tx1 = sitk.TranslationTransform(3,(1,7,76))   |
+|  tx1 = sitk.TranslationTransform(3,(1,7,76))      |  tx2 = sitk.Euler3DTransform()                 |
+|  tx2 = sitk.Euler3DTransform()                    |  tx = sitk.CompositeTransform([tx1, tx2])      |
+|  tx.AddTransform(tx1)                             |                                                |
+|  tx.AddTransform(tx2)                             |                                                |
++---------------------------------------------------+------------------------------------------------+
+
+This change does effect compatibility when reading or writing `CompositeTransforms` between version.
```
Comment (zivy) [inline docs/source/migrationGuide2.0.rst:37]: version -> versions

File: docs/source/migrationGuide2.0.rst [hunk 3]
```diff
@@ -1,34 +1,69 @@
...
-|                                                          |  grayscale_image = sitk.Cast(sitk.RescaleIntensity(0.299*sitk.VectorIndexSelectionCast(image,0,sitk.sitkFloat32) +                |
-|  image = sitk.ReadImage(file_name)                       |                                                    0.587*sitk.VectorIndexSelectionCast(image,1,sitk.sitkFloat32) +                |
-|  grayscale_image = sitk.VectorIndexSelectionCast(image,0)|                                                    0.114*sitk.VectorIndexSelectionCast(image,0,sitk.sitkFloat32)), sitk.sitkUInt8)|
-+----------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------+
+Filter's Execute Method
++++++++++++++++++++++++
 
+The Filters' `Execute` methods with filter parameters as arguments is removed. A filter object's parameters are only set via the SetXYZ methods. For example the `STAPLEImageFilter class <https://simpleitk.org/doxygen/latest/html/classitk_1_1simple_1_1STAPLEImageFilter.html>`__:
```
Comment (zivy) [inline docs/source/migrationGuide2.0.rst:43]: is removed -> has been removed.
or
is removed -> was removed.

File: docs/source/migrationGuide2.0.rst [hunk 4]
```diff
@@ -1,34 +1,69 @@
...
++-----------------------------------------------------------+---------------------------------------------------------+
+| .. code-block:: python                                    | .. code-block:: python                                  |
+|                                                           |                                                         |
+|  staple_filter = sitk.STAPLEImageFilter()                 |  staple_filter = sitk.STAPLEImageFilter()               |
+|                                                           |  staple_filter.ConfidenceWeight(confidenceWeight)       |
+|  staple_image = staple_filter.Execute(image1,             |  staple_filter.SetForegroundValue(foregroundValue)      |
+|                                       image2, image3,     |  staple_filter.SetMaximumIterations(maximumIterations)  |
+|                                       confidenceWeight,   |  staple_image = staple_filter()                         |
+|                                       foregroundValue,    |                                                         |
```
Comment (zivy) [inline docs/source/migrationGuide2.0.rst:55]: New code, line `staple_image = staple_filter()` should be changed to `staple_filter.Execute([image1, image2, image3])`

File: docs/source/migrationGuide2.0.rst [hunk 5]
```diff
@@ -38,48 +73,95 @@ The procedural interfaces for BinaryErode, BinaryDilate and BinaryClosingByRecon
 |  seg = sitk.BinaryDilate(seg, 5)                         |  seg = sitk.BinaryDilate(seg, [5]*seg.GetDimension()) |
 +----------------------------------------------------------+-------------------------------------------------------+
 
-Transformations
-+++++++++++++++
+The follow is a list of procedures affected by this change:
```
Comment (zivy) [inline docs/source/migrationGuide2.0.rst:76]: follow -> following

File: docs/source/migrationGuide2.0.rst [hunk 6]
```diff
@@ -38,48 +73,95 @@ The procedural interfaces for BinaryErode, BinaryDilate and BinaryClosingByRecon
...
+* `GrayscaleMorphologicalOpening`
+* `MorphologicalGradient`
+* `OpeningByReconstruction`
+* `WhiteTopHat`
+
+
+FastMarchingImageFilter
++++++++++++++++++++++++
+
+The `FastMarchingImageFilter::AddTrailPoints` and `FastMarchingImageFilter::ClearTrialPoints` are removed. The `SetTrailPoints` method can be used instead to set the whole list of trial points.
```
Comment (zivy) [inline docs/source/migrationGuide2.0.rst:100]: Trail -> Trial in `AddTrail` and `SetTrail`

**Review by zivy — CHANGES_REQUESTED** [review]

File: docs/source/migrationGuide2.0.rst (hunk 6 above)
Comment (blowekamp) [reply docs/source/migrationGuide2.0.rst:100]: I am not sure what you are referring to here?

Comment (zivy) [reply docs/source/migrationGuide2.0.rst:100]: The order of 'i' and 'a' is flipped in the `AddTrialPoints` and `SetTrialPoints`

**Review by zivy — APPROVED** [review]

**Pull request merged**